sphinx_needs/needs.py

```diff
@@ -454,6 +455,129 @@ def visitor_dummy(*_args, **_kwargs) -> None:
     """
     pass
 
+#todo: test isattributeset
+def isattributeset(grid: {}, attributes: []) -> bool:
```

Comment:
```suggestion
def is_attribute_set(grid: {}, attributes: []) -> bool:
```

also, missing type annotations